src/index.js

```diff
@@ -985,18 +1017,20 @@ export default function (babel) {
         }
       },
 
-      Function(path) {
-        if (path.node.kind === "constructor" || path.node.kind === "set") return;
+      Function: {
+        exit(path) {
```

Comment:
Hmm, because it is an exit visitor is maybe why you are getting `NamedArrowDeclaration`s for free...